### DISCUSSION TITLE
Develco / Frient SIRZB-110

### DISCUSSION THREAD

**Discussion by Matriciel** [Koenkk/zigbee2mqtt#20444] [Devices]
Hi,
I use 2 siren Develco SIRZB-110  witch don't have same firmware.

The first is fully working as expected with Z2M:
![ksnip_20231230-162622](https://github.com/Koenkk/zigbee2mqtt/assets/47857157/9f208a1d-f99b-455b-941b-59955264cfea)
`{
    "ac_status": false,
    "alarm": false,
    "battery": 100,
    "battery_low": false,
    "enrolled": true,
    "ias_cie_address": "0x00124b00256c4ca5",
    "linkquality": 87,
    "max_duration": 900,
    "restore_reports": true,
    "supervision_reports": true,
    "tamper": false,
    "test": false,
    "voltage": 4100,
    "zone_id": 23,
    "device": {
        "dateCode": "2022-04-20 23:13",
        "friendlyName": "0x0015bc0041001c4d",
        "hardwareVersion": "5.0.2",
        "ieeeAddr": "0x0015bc0041001c4d",
        "manufacturerID": 4117,
        "manufacturerName": "frient A/S",
        "model": "SIRZB-110",
        "networkAddress": 65244,
        "powerSource": "Mains (single phase)",
        "softwareBuildID": "1.7.1",
        "type": "Router",
        "zclVersion": 3
    }
}`

The second is not running as espected, some mode are not usable.
![ksnip_20231230-162727](https://github.com/Koenkk/zigbee2mqtt/assets/47857157/a8005a2c-9e86-4bdb-b920-b982d0e052b0)
`{
    "ac_status": false,
    "alarm": false,
    "battery": 100,
    "battery_low": false,
    "enrolled": true,
    "ias_cie_address": "0x00124b00256c4ca5",
    "linkquality": 76,
    "max_duration": 68,
    "restore_reports": true,
    "supervision_reports": true,
    "tamper": false,
    "test": false,
    "voltage": 4000,
    "zone_id": 23,
    "device": {
        "dateCode": "2022-04-25 01:40",
        "friendlyName": "0x0015bc004100171a",
        "hardwareVersion": "5.0.2",
        "ieeeAddr": "0x0015bc004100171a",
        "manufacturerID": 4117,
        "manufacturerName": "frient A/S",
        "model": "SIRZB-110",
        "networkAddress": 12121,
        "powerSource": "Mains (single phase)",
        "softwareBuildID": "1.9.3",
        "type": "EndDevice",
        "zclVersion": 7
    }
}`

I have found that endpoint 43 was differents. With Zigbee Jeedom, i have modified endpoint like this:

Running one:
`43::ias_wd::start_warning::1::1::1::#duration#::1::1|Son BAS, Type URGENCE; 43::ias_wd::start_warning::5::1::1::#duration#::1::1|Son MOYEN, Type URGENCE; 43::ias_wd::start_warning::9::1::1::#duration#::1::1|Son FORT, Type URGENCE; 43::ias_wd::start_warning::1::1::2::#duration#::1::1|Son BAS, Type INCENDIE; 43::ias_wd::start_warning::5::1::2::#duration#::1::1|Son MOYEN, Type INCENDIE; 43::ias_wd::start_warning::9::1::2::#duration#::1::1|Son FORT, Type INCENDIE; 43::ias_wd::start_warning::1::1::4::#duration#::1::1|Son BAS, Type INTRUSION; 43::ias_wd::start_warning::5::1::4::#duration#::1::1|Son MOYEN, Type INTRUSION; 43::ias_wd::start_warning::9::1::4::#duration#::1::1|Son FORT, Type INTRUSION`

The new version :
`43::ias_wd::start_warning::5::1::1::#duration#::1::1|Son BAS, Type URGENCE; 43::ias_wd::start_warning::5::1::2::#duration#::1::1|Son MOYEN, Type URGENCE; 43::ias_wd::start_warning::5::1::3::#duration#::1::1|Son FORT, Type URGENCE; 43::ias_wd::start_warning::3::1::1::#duration#::1::1|Son BAS, Type INCENDIE; 43::ias_wd::start_warning::3::1::2::#duration#::1::1|Son MOYEN, Type INCENDIE; 43::ias_wd::start_warning::3::1::3::#duration#::1::1|Son FORT, Type INCENDIE; 43::ias_wd::start_warning::4::1::1::#duration#::1::1|Son BAS, Type INTRUSION; 43::ias_wd::start_warning::4::1::2::#duration#::1::1|Son MOYEN, Type INTRUSION; 43::ias_wd::start_warning::4::1::3::#duration#::1::1|Son FORT, Type INTRUSION
`

I don't know how create a new converter or add this update to device database.

**Comment by Aeliassen** [comment]
I just got mine and I'm running firmware 1.9.3. When I trigger the siren with MQTT command
{"warning":{"duration":60,"level":"low","mode":"burglar","strobe":true,"strobe_duty_cycle":0}} it gives an sound for 60 secounds, but if I change mode to fire, emergency or other it makes the same sound. If I change level to medium it makes another sound, but the same volume. 

Is this what you are experiencing?

**Reply by Matriciel** [reply]
Hi, yes it is that problem.
I have changed endpoint configuration in the Zigbee Jeedom like previous post but i don't know how make modification in Z2M and how identify the different siren models

**Reply by Aeliassen** [reply]
Have you tried to find a way to just downgrade the firmware to 1.7.1 on the siren. Maybe this can help get the siren working again?

**Reply by Matriciel** [reply]
I don't know how change the firmware on theses devices ...

**Reply by LMQTLV** [reply]
Hello,
I have exactly the same problem with a Frient one also with firmware 1.9.3 :-(
Does anyone know a way to downgrade it to 1.7.1 please ?

**Reply by CannyGent** [reply]
Hi - i just got one of these and firmware 2.0.3 was installed... did anybody find away to downgrade to 1.7.1?

**Reply by LMQTLV** [reply]
@Matriciel 
> I don't know how change the firmware on theses devices ...

@CannyGent 
> Hi - i just got one of these and firmware 2.0.3 was installed... did anybody find away to downgrade to 1.7.1?

If you are still looking for a way to downgrade the firmware of this siren, I found a way about a year ago and posted a tutorial here (in French though, sorry for English speakers 😉 ) :
[Voici le processus complet :](https://community.jeedom.com/t/sirene-frient-plusieurs-versions/107048/7)

(and firmware file v1.7.1 is two posts above)

**Reply by CannyGent** [reply]
Thanks will have a look at that and report how I get on... so far have been trying to get the z2m ota functionality to push the firmware but no success. The siren is not recognised by z2m as supporting ota updates.

**Comment by CannyGent** [comment]
Hello all - do you think there will be a way to progress the Frient SIRZB-110 and get compatibility with the newer firmwares?

**Reply by LMQTLV** [reply]
Hello,
For my part, given the apparent lack of interest of the Z2M developers for sirens despite the hundreds of topics reporting this problem, and this for the numerous models of sirens which have the same problem, I have given up hope of having native compatibility under Z2M 😢

**Reply by CannyGent** [reply]
Yeah - seems that way. It is hard to understand because Alarmo is great and there is a need for a good zigbee siren with squawk and alarm functions and the Frient looks perfect. I have written to Develco to see if they could share the updated specification which seems to apply to firmwares beyond 1.7.1 as theoretically it should then be a fairly easy job to update zigbee2mqtt to support these devices.

**Reply by Aeliassen** [reply]
> I have written to Develco to see if they could share the updated specification which seems to apply to firmwares beyond 1.7.1

You should try to write to Frient too. I asked both for old firmware and only Frient replied. They were super helpfull.

**Reply by CannyGent** [reply]
I can't find a contact for Frient... do you still have the firmware file to share? Thanks

**Reply by Matriciel** [reply]
I have the firmware and you can download it here -> https://www.le-matriciel.cloud/index.php/s/g9L2wprJDXJDr2Y

**Reply by CannyGent** [reply]
Thanks!  I think that is same file I found elsewhere but z2m not interested in downgrading with it using the custom OTA method... I am going to try the ZHA route that @LMQTLV posted about above and hopefully that will get me a working siren!

**Reply by Matriciel** [reply]
I have also downgraded my four device with an old Home Assistant with Zigpy on a temporary docker.

**Reply by chris-1243** [reply]
> Hello,
> 
> For my part, given the apparent lack of interest of the Z2M developers for sirens despite the hundreds of topics reporting this problem, and this for the numerous models of sirens which have the same problem, I have given up hope of having native compatibility under Z2M 😢 

I do not think it is a lack of interest but lack of time. Z2M is a nice project, time consumming and open source.
Keep also in mind, devs can not have any zigbee object to test and try. This is where the community is a huge help. It is our job to bring support for new devices as much as we can

**Reply by LMQTLV** [reply]
Hmmm, I don't know ...

As I said :
> despite the hundreds of topics reporting this problem, and this for the numerous models of sirens which have the same problem

**Reply by CannyGent** [reply]
> Hello all - do you think there will be a way to progress the Frient SIRZB-110 and get compatibility with the newer firmwares?

Ok troops - thanks for all the help... downgrade was successful using older HA Docker with ZHA method as described [here](https://community.jeedom.com/t/sirene-frient-plusieurs-versions/107048/7) 

Rejoined it to z2m and strangely it still reports firmware 2.0.3 but it is definitely downgraded because it now works great.

**Reply by LMQTLV** [reply]
I'm glad I could help you ☺️
